src/main/java/com/unb/tracker/web/CourseController.java

```diff
@@ -185,15 +187,31 @@ public String courseSave(@ModelAttribute Course course, ModelMap map, Principal
 
         if (course.getSection().isEmpty()) {
             course.setSection("");
-            courseRepository.save(course);
-
-            map.addAttribute("course", course);
-            return "redirect:/" + user.getUsername() + "/" + course.getName();
         }
 
         courseRepository.save(course);
+        
+        if(newCourse) {
```

Comment:
This is a weird fix to the issue we were having but it works. 

Basically, the seats weren't being initialized which was causing an error. This fix involves initializing the seats, but it has to be done after the course is saved which is why it is placed here and uses a `newCourse` flag